tests/ingest_runs/test_genomic_data_loader.py

```diff
@@ -0,0 +1,299 @@
+        entity_type = args[0]
+        loader = LoadStage(
+            TARGET_API_CONFIG,
+            settings.DATASERVICE_URL,
+            [entity_type],
+            study_generator.study_id,
+            cache_dir=os.path.join(tmpdir, "temp"),
+            dry_run=True,
+        )
+        loader.run({entity_type: args[1]})
+        return loader.uid_cache
+
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "_load_entities",
+        side_effect=_dry_load_entities
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_load_bs_gf_links(mocker):
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "load_specimen_harmonized_gf_links"
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_load_se_gf_links(mocker):
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "load_seq_exp_harmonized_genomic_files"
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_validate(mocker):
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "_validate_load"
+    )
+    return mock
+
+
+def test_load_harmonized_genomic_files(
+    mock_s3_scrape, study_generator, mock_load_entities
+):
+    """
+    Test the loading of the harmonized genomic files from the genomic workflow
+    output manifest.
+    """
+    loader = GenomicDataLoader(FAKE_STUDY)
+    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
+    loader.load_harmonized_genomic_files(manifest_df)
+    mock_s3_scrape.assert_called_once()
+    mock_load_entities.assert_called_once()
+    entity_type, df = mock_load_entities.call_args_list[0][0]
+    assert isinstance(df, pd.DataFrame)
+    assert entity_type == GEN_FILE
+    assert not df.empty
+    manifest_df = manifest_df.drop_duplicates(["KF Participant ID"])
+    assert manifest_df.shape[0] == df.shape[0]
+    # Check that the columns are as expected
+    for column in GF_EXPECTED_COLUMNS:
+        assert column in df
+    assert df.shape[1] == len(GF_EXPECTED_COLUMNS)
+
+
+def test_load_specimen_harmonized_gf_links(
+    study_generator, mock_s3_scrape, mock_load_entities
+):
+    """
+    Test the load_specimen_harmonized_gf_links function. This is the second
+    step of the ingestion process.
+    """
+    loader = GenomicDataLoader(FAKE_STUDY)
+    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
+    gfs = loader.load_harmonized_genomic_files(manifest_df)
+    loader.load_specimen_harmonized_gf_links(gfs)
+    mock_s3_scrape.assert_called_once()
+    assert mock_load_entities.call_count == 2
+    # Now check the biospecimen gf links loading
+    entity_type2, df2 = mock_load_entities.call_args_list[1][0]
+    assert entity_type2 == BIO_GEN_FILE
+    assert isinstance(df2, pd.DataFrame)
+    assert not df2.empty
+    manifest_df = manifest_df.drop_duplicates(["KF Participant ID"])
```

Comment:
Same comment as above. In this test we want to check that every harmonized gf got linked to one unique specimen (since these harmonized files represent a re-aligned version of the single specimen source sequencing file). So if we have 30 harmonized files then we should have 30 unique bs-gf links

Reply:
I agree. But we only have 10 harmonized files, right?